mobile/src/Student/containers/GameSessionContainer.js

```diff
@@ -160,23 +96,82 @@ const GameSessionContainer = ({ children }) => {
             })
     }
 
-    useEffect(() => {
-        if (isNullOrUndefined(gameSession)) {
-            clearStorage()
+    const handleSubscribeToGame = (gameSession) => {
+          if (isNullOrUndefined(gameSession)) {
+            resetState()
             return
-        }
-
-        storeDataToLocalStorage(localStorageKeys.gameCode, `${gameSession.gameCode}`)
-
-        let subscription =
+          }
+          
+          let subscription =
             global.apiClient.subscribeUpdateGameSession(
                 gameSession.id,
                 (gameSessionResponse) => {
                     setGameSession(gameSessionResponse)
                 })
+    }
+
+    const handleAddTeam = async (teamName) => 
+    {
+      return global.apiClient
+            .addTeamToGameSessionId(gameSession.id, teamName, null)
+            .then((team) => {
+                console.debug(team)
+                if (!team) {
+                    console.error("Failed to add team")
+                    return
+                }
+
+
+                global.apiClient
+                    .addTeamMemberToTeam(team.id, true, uuid.v4())
+                    .then((teamMember) => {
+                        if (!teamMember) {
+                            console.error("Failed to add team member")
+                            return
+                        }
+
+                        if (isNullOrUndefined(team.teamMembers)) {
+                            team.teamMembers = [teamMember]
+                        }
+
+                        return setTeamInfo(team, teamMember)
+                    }).catch((error) => {
+                        console.error(error)
+                    })
+            })
+            .catch((error) => {
+                console.error(error)
+            })
+    }
+
+    const handleAddTeamAnswer = async (question, answer, gameSessionState) =>
+    {
+      return  (global.apiClient
+                  .addTeamAnswer(
+                      teamMember.id,
+                      question.id,
+                      answer.text,
+                      (gameSessionState === GameSessionState.CHOOSE_CORRECT_ANSWER && true || gameSessionState === GameSessionState.CHOOSE_TRICKIEST_ANSWER && false),
+                      (gameSessionState === GameSessionState.CHOOSE_CORRECT_ANSWER && false || gameSessionState === GameSessionState.CHOOSE_TRICKIEST_ANSWER && true)
```

Comment:
A bit confused here, what's the purpose of `&& true` and `&& false`

Reply:
Good point. Rewritten for clarity using ternary. the goal here is to ensure that on `CHOOSE_CORRECT_ANSWER`, `isChosen = true` and `isTrickAnswer = false,` while on `CHOOSE_TRICKIEST_ANSWER`, `isChosen=false` and `isTrickAnswer = true`